### ISSUE TITLE
On Redhat, lost data with Unison

### ISSUE THREAD

**Issue by lieh1203** [bcpierce00/unison#618]
Hi,
I performed two-way file synchronization on the directory named "uploads" between the two machines. Machine A is the production server, and Machine B is a disaster recovery server.
Usually the website of machine B is stop, and there is no operation to delete files in the system.
The website has been online for more than a year. But recently, I found that some files in this directory were randomly deleted every month, I’m not sure what the problem is.
I checked audit.log on Redhat and found that there were some deletion operations in it. I suspect that the problem is with Unison, because only Unison is listening to this directory, so I think it is less likely that other software has a problem
Related configuration on the below

audit.log
 
![aa](https://user-images.githubusercontent.com/34055613/147647711-4325c9cf-5338-42c1-b514-6067659c213b.png)

default.prf

A Machine config:

    root = /app/appcis/uploads/
    root = ssh://cisapp@IP of Machine B//app/appcis/uploads/
    batch = true 
    maxthreads = 300
    owner = true
    group = true 
    perms = -1 
    fastcheck = false 
    rsync = false 
    sshargs = -C 
    xferbycopying = true 
    confirmbigdel = false 
    log = true
    logfile = /app/appcis/unison-2.51.2/unison.log

B Machine config:

    root = /app/appcis/uploads/
    root = ssh://cisapp@IP of Machine A//app/appcis/uploads/
    batch = true 
    maxthreads = 300
    owner = true
    group = true 
    perms = -1 
    fastcheck = false 
    rsync = false 
    sshargs = -C 
    xferbycopying = true 
    confirmbigdel = false 
    log = true
    logfile = /app/appcis/unison-2.51.2/unison.log

unison.sh
    #/bin/bash
    src="/app/appcis/uploads/" 
    /app/appcis/inotify-tools-3.14/bin/inotifywait -mrq -e create,delete,modify,move $src | while read line; do
    /app/appcis/unison-2.51.2/bin/unison -servercmd=/app/appcis/unison-2.51.2/bin/unison
    echo -n "$(date +%F-%T) $line" >> /app/appcis/inotify-tools-3.14/inotify.log
    done

I am searching for a long time on net. But no solution.

Thanks, Any help would be appreciated.

**Comment by tleedjarv** [comment]
Is there any mention of the affected files in `unison.log`?

**Comment by lieh1203** [comment]
> 有没有提到受影响的文件`unison.log`？
thanks reply.
The highlight is a missing file, However, some missing files are not recorded in the log.

![bccc9d231ef322c061a53b0e9880c29](https://user-images.githubusercontent.com/34055613/147674174-9e597355-9037-4113-80d7-aec9f2abf6ea.jpg)

**Comment by lieh1203** [comment]
> Please read the comment in #616 where I referred you to how to debug, how to provide a proper bug report, and how to use the mailinglist for help.

is this email that unison-users<unison-users+subconfirm@seas.upenn.edu>? 
I tried to join this groups but not reply me.

**Comment by gdt** [comment]
Yes, that's right.  It might take a few days for someone to process it.

Please read carefully https://github.com/bcpierce00/unison/wiki/Reporting-Bugs-and-Feature-Requests and follow all the steps.  For a bug report, you will have to show that it's a bug, which is much more than something bad happening and guessing it is unison.   I don't mean to claim that unison has no bugs -- just that the path to fixing them comes from having a minimal reproduction recipe.   You haven't explained versions, you haven't explained if you are using the watcher, you haven't reduce to running unison once at a time without the watcher, and you haven't written scripts to take a census of both sides before and after.

**Comment by lieh1203** [comment]
> Yes, that's right. It might take a few days for someone to process it.
> 
> Please read carefully https://github.com/bcpierce00/unison/wiki/Reporting-Bugs-and-Feature-Requests and follow all the steps. For a bug report, you will have to show that it's a bug, which is much more than something bad happening and guessing it is unison. I don't mean to claim that unison has no bugs -- just that the path to fixing them comes from having a minimal reproduction recipe. You haven't explained versions, you haven't explained if you are using the watcher, you haven't reduce to running unison once at a time without the watcher, and you haven't written scripts to take a census of both sides before and after.

@gdt 
Thanks, I'll carefully check debug information again.

**Comment by gdt** [comment]
timed out...

Feel free to reopen or start over if you get good debugging data.
